### PR TITLE
James 3693 update redis config to support read from replicas

### PR DESCRIPTION
**Pull request by hungphan227**
https://github.com/linagora/james-project/issues/5172

RedisRateLimiter is not updated because I see that ratelimit library mostly run redisScriptingReactiveCommands.evalsha which has to work with master node (this method is used to run lua script).

### CODE REVIEW

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc [hunk 1]
```diff
@@ -18,6 +18,9 @@ to get some examples and hints.
 | redis.topology
 | Redis server topology. Defaults to standalone. Possible values: standalone, cluster, master-replica
 
+| redis.readFrom
+| The property to determine how Lettuce routes read operations to Redis server with topologies other than standalone. Defaults to master. Possible values: master, masterPreferred, replica, replicaPreferred, any
```
Comment (quantranhong1999) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc:22]: add a reference to lettuce values IMO

File: backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisConfigurationTest.scala [hunk 2]
```diff
@@ -48,6 +49,7 @@ class RedisConfigurationTest extends AnyFlatSpec with Matchers {
     config.setListDelimiterHandler(new DefaultListDelimiterHandler(','))
     config.addProperty("redisURL", "redis://localhost:6379,redis://localhost:6380")
     config.addProperty("redis.topology", "cluster")
+    config.addProperty("redis.readFrom", "any")
```
Comment (quantranhong1999) [inline backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisConfigurationTest.scala:52]: Can we have a test that `readFrom=invalid` ?

File: backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisMasterReplicaExtension.java [hunk 3]
```diff
@@ -137,17 +135,6 @@ public void beforeEach(ExtensionContext extensionContext) throws Exception {
...
-    @Override
-    public Module getModule() {
-        return new AbstractModule() {
-            @Provides
-            @Singleton
-            public RedisConfiguration provideRedisConfiguration() {
-                return redisClusterContainer.getRedisConfiguration();
-            }
-        };
-    }
```
Comment (quantranhong1999) [inline backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisMasterReplicaExtension.java:149]: Q: why remove this now?

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc (hunk 1 above)
Comment (quantranhong1999) [reply server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc:22]: BTW can you add this entry in an example `redis.properties` too?

**Review by Arsnael — COMMENTED** [review]
Read it, nothing to add

**Comment by Arsnael** [comment]
```
11:11:02,443 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.3.0:check (check-style) on project apache-james-backends-redis: You have 2 Checkstyle violations. -> [Help 1]
```

File: backends-common/redis/src/main/java/org/apache/james/backends/redis/RedisConfiguration.scala [hunk 4]
```diff
@@ -96,4 +106,4 @@ case object Cluster extends RedisTopology
 
 case object MasterReplica extends RedisTopology
 
-case class RedisConfiguration(redisURI: RedisUris, redisTopology: RedisTopology, ioThreads: Option[Int], workerThreads:Option[Int])
+case class RedisConfiguration(redisURI: RedisUris, redisTopology: RedisTopology, readFrom: ReadFrom, ioThreads: Option[Int], workerThreads:Option[Int])
```
Comment (vttranlina) [inline backends-common/redis/src/main/java/org/apache/james/backends/redis/RedisConfiguration.scala:109]: I don't see where we use `readFrom` in prod code

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc (hunk 1 above)
Comment (hungphan227) [reply server/apps/distributed-app/docs/modules/ROOT/pages/configure/redis.adoc:22]: done

File: backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisConfigurationTest.scala (hunk 2 above)
Comment (hungphan227) [reply backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisConfigurationTest.scala]: done

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**